devel/MacOSX.sdk/Portfile

```diff
@@ -607,26 +607,30 @@ if {${subport} eq ${name}} {
         move ${worksrcpath}/[lindex ${extract_paths} 0] ${destroot}${sdk_dir}/${subport}
     }
 
-    if {[vercmp ${sdk_version} >= 11.0]} {
-        supported_archs arm64 x86_64
-    } elseif {[vercmp ${sdk_version} >= 10.14]} {
-        supported_archs x86_64
-    } elseif {[vercmp ${sdk_version} >= 10.7]} {
-        supported_archs i386 x86_64
-    } elseif {[vercmp ${sdk_version} >= 10.6]} {
-        supported_archs i386 ppc x86_64
-    } elseif {[vercmp ${sdk_version} >= 10.4]} {
-        supported_archs i386 ppc x86_64 ppc64
-    } else {
-        supported_archs ppc
-    }
-
-    if {[llength ${supported_archs}] > 1} {
-        configure.universal_archs {*}${supported_archs}
-        variant universal {}
-        default_variants +universal
-        variant_set universal
-    }
+    # limiting `supported_archs` might prevent SDK from being installed even if it is useful
+    #
+    #if {[vercmp ${sdk_version} >= 11.0]} {
+    #    supported_archs arm64 x86_64
+    #} elseif {[vercmp ${sdk_version} >= 10.14]} {
+    #    supported_archs x86_64
+    #} elseif {[vercmp ${sdk_version} >= 10.7]} {
+    #    supported_archs i386 x86_64
+    #} elseif {[vercmp ${sdk_version} >= 10.6]} {
+    #    supported_archs i386 ppc x86_64
+    #} elseif {[vercmp ${sdk_version} >= 10.4]} {
+    #    supported_archs i386 ppc x86_64 ppc64
+    #} else {
+    #    supported_archs ppc
+    #}
+    #
+    #if {[llength ${supported_archs}] > 1} {
+    #    configure.universal_archs {*}${supported_archs}
+    #    variant universal {}
+    #    default_variants +universal
+    #    variant_set universal
+    #}
+    #
+    supported_archs     noarch
```

Comment:
This is certainly inaccurate but this port is a very special case so maybe this is an acceptable simplification.

Do you know if this would do anything to fix https://trac.macports.org/ticket/66918? Does rev-upgrade skip scanning ports marked noarch? If so, that would be an argument in favor of this change.

Reply:
> This is certainly inaccurate but this port is a very special case so maybe this is an acceptable simplification.

I would be happy to revert the change.
Setting `configure.cxx_stdlib` to empty is similarly inaccurate.
Should `configure.cxx_stdlib` also be set to either libc++ or libstdc++ depending on the SDK version?

> Do you know if this would do anything to fix https://trac.macports.org/ticket/66918? Does rev-upgrade skip scanning ports marked `noarch`? If so, that would be an argument in favor of this change.

I am afraid the rev-upgrade issue persists even with the change.

Reply:
> I would be happy to revert the change.

I'm not saying yet that we should revert it. It certainly would reduce the number of systems on which the port needs to build, and the archives that get produced would indeed be the same.

> Setting `configure.cxx_stdlib` to empty is similarly inaccurate.

That is true… the comment in the port explains that this was done to eliminate an unnecessary clang dependency. If there is a better solution to that, that's fine.

> Should `configure.cxx_stdlib` also be set to either libc++ or libstdc++ depending on the SDK version?

I'm not sure if that would have the same effect.

Reply:
> Does rev-upgrade skip scanning ports marked `noarch`? 

No, and that's probably because it also scans for missing files. This issue will require a change in "base", I'm afraid. But maybe that can be as simple as making `noarch` skip the registration of binary files. IIRC this is basically what I did to have a semi-functional `rev-upgrade` feature in my LinuxPorts adaptation (I made `isBinary()` a stub returning false unconditionally so).

This would make some sort of sense ... treating all files installed by a port claiming that it has only "noarch" files as if they were not "binary" even if some of them would actually fit that description. I think it can be argued that no port setting this expects or should expect to have its files checked for shared library dependencies.

The alternative would be a new `rev_upgrade_skip_archceck` option.

Reply:
> This issue will require a change in "base", I'm afraid.

Or an unorthodox approach: create a nicely compressed DMG of the SDK directory and install that, plus a script that handles the mounting so the user doesn't have to remember where the DMG is installed. Ports that get their content off a DMG know how to mount them on custom mountpoint not under `/Volumes` so that script could make the DMG content appear under the same path as where it's found now.

I think there are additional advantages to this; these SDKs install a really large number of (small) files which make them potentially SSD-unfriendly and costly for both the registry and the rev-upgrade scan. Is there any indication that people who install these ports actually use them more than occasionally?

Reply:
Fixing the rev-upgrade issue would definitely be nice.
However, as you point out, it must either be fixed in the base or in the Portfile with an "unorthodox approach."
The ideas are interesting, but I think they are a bit outside the scope of this PR.

Reply:
> create a nicely compressed DMG of the SDK directory and install that

Yuck!

> I suppose a startupitem could be provided for those who do, so the DMG gets mounted when they log in?

Eww!